src/smtencoding/FStar.SMTEncoding.Z3.fs

```diff
@@ -546,26 +546,22 @@ let mk_input theory =
                 BU.prefix_until (function CheckSat -> true | _ -> false) |>
                 Option.get
             in
-            let pp        = List.map (declToSmt options) in
-            let pp_no_cap = List.map (declToSmt_no_caps options) in
+            let pp = List.map (declToSmt options) in
             let suffix = check_sat::suffix in
             let ps_lines = pp prefix in
             let ss_lines = pp suffix in
             let ps = String.concat "\n" ps_lines in
             let ss = String.concat "\n" ss_lines in
 
             (* Ignore captions AND ranges when hashing, otherwise we depend on file names *)
-            let uncaption = function
-            | Caption _ -> Caption ""
-            | Assume a -> Assume ({ a with assumption_caption = None })
-            | DeclFun (n, a, s, _) -> DeclFun (n, a, s, None)
-            | DefineFun (n, a, s, b, _) -> DefineFun (n, a, s, b, None)
-            | d -> d
+            let hs =
+              if Options.log_queries()
+              && Options.keep_query_captions ()
+              then prefix
+                   |> List.map (declToSmt_no_caps options)
+                   |> String.concat "\n"
+              else ps
```

Comment:
Looks fine to me. A minor nit: we could move `Options.log_queries () && Options.keep_query_options ()` to a single function `print_query_captions` or something, and use it here as well as in the definition of `declToSMT`.

Reply:
One way to check correctness here is that at this point, `ps = prefix  |> List.map (declToSmt options) |> String.concat "\n"`. And when `Options.log_queries () && Options.keep_query_options () = false`, `declToSmt = declToSmt_no_caps`.

Reply:
I didn't understand your second remark, @aseemr. Can you say more, please?

Reply:
Oh nothing much, was just trying to convince myself that in the `else` branch, `ps` is already same as `prefix |> List.map (declToSmt_no_caps options) |> String.concat "\n"`.